customTests/azure_nccl_allreduce.nhc

```diff
@@ -12,18 +11,42 @@ function collect_nccl_allreduce_data() {
    NCCL_ARGS="-b $MESSAGE_SIZE -f 2 -g 1 -e $MESSAGE_SIZE -c 1"
 
    # ensure unloaded before use
-   source /etc/profile.d/modules.sh && module -s unload mpi/hpcx
-   nccl_allreduce_out=$(source /etc/profile.d/modules.sh && module -s load mpi/hpcx && mpirun $MPI_ARGS $ENVIRON_VARS /opt/nccl-tests/build/all_reduce_perf $NCCL_ARGS)
+   MPI_ARGS="-np 8 --map-by ppr:8:node -bind-to numa -mca coll_hcoll_enable 0 --allow-run-as-root"
+   nccl_allreduce_out=$(mpirun $MPI_ARGS $ENVIRON_VARS $ALL_REDUCE_PATH $NCCL_ARGS)
    nccl_allreduce_out_rc=$?
    if [[ $nccl_allreduce_out_rc != 0 ]]; then
-      log "nccl_allreduce_freq_out"
       die 1 "$FUNCNAME: nccl_allreduce returned error code $nccl_allreduce_out_rc"
    fi
    IFS=$'\n'
    nccl_allreduce_out_lines=( $nccl_allreduce_out )
    IFS=$' \t\n'
 }
 
+function check_dependencies() {
+   if [[ ! -f $ALL_REDUCE_PATH ]]; then
+      die 1 "$FUNCNAME: all reduce executable path: $ALL_REDUCE_PATH not found. Ensure nccl-tests is installed and built in."
+      return 1
+   fi
+   # attempt to source HPCX modules
+   if [ -f '/etc/profile.d/modules.sh' ] && [ -z $HPCX_DIR ] ;then
```

Comment:
Where does this variable come from?

Reply:
This variable is sourced when ever HPCx is initialized:
- loaded by the module
- loaded by sourcing the hpcx init script (/opt/hpcx-v2.16-gcc-mlnx_ofed-ubuntu22.04-cuda12-gdrcopy2-nccl2.18-x86_64/hpcx-init.sh)